test/playwright/utils/navigation.ts

```diff
@@ -90,7 +121,100 @@ export const currentContentType = async (page: Page) => {
 }
 
 export const dismissTranslationBanner = async (page: Page) => {
-  await page
-    .locator('[data-testid="banner-translation-ar"] [aria-label="Close"]')
-    .click()
+  await page.locator('[data-testid="banner-translation"] button').click()
+}
+
+export const goToSearchTerm = async (
+  page: Page,
+  term: string,
+  searchType: SearchType = ALL_MEDIA,
+  options: {
+    mode?: RenderMode
+    dir?: 'ltr' | 'rtl'
+    query?: string // Only for SSR mode
+  } = {}
+) => {
+  const dir = options.dir || 'ltr'
+  const mode = options.mode ?? 'SSR'
+  const query = options.query ? `&${options.query}` : ''
+
+  const prefix = dir === 'ltr' ? '' : '/ar'
+  if (mode === 'SSR') {
+    await page.goto(
+      `${prefix}/search/${searchTypePath(searchType)}?q=${term}${query}`
+    )
+    if (dir === 'rtl') {
+      await dismissTranslationBanner(page)
+    }
+  } else {
+    await page.goto(`/${prefix}`)
+    if (dir === 'rtl') {
+      await dismissTranslationBanner(page)
+    }
+    // Select the search type
+    if (searchType !== 'all') {
+      await page.click('[aria-label="All content"]')
+      await page.click(
+        `button[role="radio"]:has-text("${searchTypeNames[searchType]}")`
+      )
+    }
+    // Type search term
+    const searchInput = page.locator('main input[type="search"]')
+    await searchInput.type(term)
+    // Click search button
+    // Wait for navigation
+    await Promise.all([
+      page.click('[aria-label="Search"]'),
+      page.waitForNavigation(),
+    ])
+    await scrollDownAndUp(page)
+  }
+}
+
+/**
+ * Click on the first <mediaType> result: a link that contains
+ * /<mediaType>/ in its URL. We cannot use the 'startsWith' `^` matcher
+ * because localized routes start with the locale prefix (e.g. /ar/image/).
+ * Scroll down and up to load all lazy-loaded content.
+ */
+export const openFirstResult = async (page: Page, mediaType: MediaType) => {
+  await page.locator(`a[href*="/${mediaType}/"]`).first().click()
+  await scrollDownAndUp(page)
+}
+
+/**
+ * Click on the first <mediaType> related result: a link that contains
+ * /<mediaType>/ in its URL in the 'aside' element for related media.
+ * We cannot use the 'startsWith' `^` matcher because localized routes
+ * start with the locale prefix (e.g. /ar/image/).
+ * Scroll down and up to load all lazy-loaded content.
+ */
+export const openFirstRelatedResult = async (
+  page: Page,
+  mediaType: MediaType
+) => {
+  await page.locator(`aside a[href*="/${mediaType}/"]`).first().click()
+  await scrollDownAndUp(page)
+}
+
+export const scrollToBottom = async (page: Page) => {
+  await page.evaluate(() => {
+    window.scrollTo(0, document.body.scrollHeight)
+  })
+  await sleep(300)
+}
+
+export const scrollToTop = async (page: Page) => {
+  await page.evaluate(() => {
+    window.scrollTo(0, 0)
+  })
+  await sleep(200)
+}
```

Comment:
Rather than these semi-arbitrary sleep times, could we poll the `clientY` on the window until it matches the expected value?

```suggestion
const waitForScrollPosition = async (toY: number) => new Promise((resolve) => {
  const interval = setInterval(() => {
    page.evaluate(() => document.clientY === toY).then((isScrolledToY) => {
      if (isScrolledToY) {
        resolve()
        clearInterval(interval)
      }
    })
  }, 50)
})

export const scrollToBottom = async (page: Page) => {
  const bottomY = await page.evaluate(() => document.body.scrollHeight)
  await page.evaluate((y) => {
    window.scrollTo(0, y)
  }, [bottomY])
  await waitForScrollPosition(bottomY)
}

export const scrollToTop = async (page: Page) => {
  await page.evaluate(() => {
    window.scrollTo(0, 0)
  })
  await waitForScrollPosition(0)
}
```

I'm writing the `page.evaluate` stuff from memory so I don't think this exact code will work but it's the general idea I wanted to propose more than anything. We'd also want to implement the playwright timeout for maximum wait times as well :wink:

Reply:
I have written some tests following the paths you've outlined in https://github.com/WordPress/openverse-frontend/issues/1244, @sarayourfriend , and I had a lot of problems with the page being scrolled to top, but still using the styles as if it wasn't. Namely, the header buttons would not have borders, or the logo Openverse text would not display. 

I wonder if we should keep the arbitrary wait times for now until we can test if this approach works well or not? I don't want to add the tests themselves here because this PR is already too big...

Reply:
Sure, we can. Maybe let's open another issue to remove them. I'm concerned that they'll be (a) a source of flakiness and (b) unnecessarily slow down tests waiting for things that have already settled (even if only a little bit).